client/components/data/query-canonical-theme/index.jsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import QueryTheme from 'components/data/query-theme';
+import { isWpcomTheme, isWporgTheme } from 'state/themes/selectors';
+
+const QueryCanonicalTheme = ( { siteId, themeId, isWpcom, isWporg } ) => (
+	<div>
+		<QueryTheme themeId={ themeId } siteId="wpcom" />
```

Comment:
Hmm. My take here was that normally, we wouldn't know initially, so `isWpcom` (and `isWporg`) would default to falsey; and even after rendering the query component, I thought it wouldn't hurt (and maybe make for a stabler behavior) to keep them around. The individual `QueryTheme` components won't re-render as long as their props don't change, so they shouldn't impact performance negatively.

Maybe we really need to check `getCanonicalTheme`, and bail early if it returns falsey? :thinking: